performance/memory-optimize-udtt.sql

```diff
@@ -140,19 +139,19 @@ open A; while 1=1 begin
 	end; close B; deallocate B
 
 	-- define indexes
-	declare @index_id int, @is_primary_key bit
+	declare @index_id int, @is_primary_key bit, @name nvarchar(256)
 	declare C cursor local fast_forward for
-		select index_id, is_primary_key
+		select index_id, is_primary_key, name
 		from sys.indexes
 		where object_id=@object_id and name is not null
 		order by index_id
 	open C; while 1=1 begin
-		fetch next from C into @index_id, @is_primary_key
+		fetch next from C into @index_id, @is_primary_key, @name
 		if @@FETCH_STATUS<>0 break
 
 		select @sql = @sql + N'
 	' +
-			case when @is_primary_key=1 then N'primary key nonclustered(' else N'nonclustered index(' end
+			case when @is_primary_key=1 then N'primary key nonclustered(' else N'index ' + @name + ' nonclustered(' end

```

Comment:
```suggestion
			case when @is_primary_key=1 then N'primary key nonclustered(' else N'index ' + quotename(@name) + ' nonclustered(' end
```

Reply:
Although I'm pretty sure we don't explicitly name nay of our UDTT PK's, and I'm not sure there's any benefit to preserving the PK name.

Reply:
This is for a non PK index case

Reply:
From what I know, non primary key indexes must have a name. And our prior filtering excludes any without a name anyway. But if you see any flaw in that thinking please let me know.

I made the other changes, thanks for taking a look!